hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusUpdateTopicHandler.java

```diff
@@ -305,6 +345,55 @@ private void validateMaybeNewSubmitKey(
         }
     }
 
+    private void validateMaybeNewFeeScheduleKey(
+            @NonNull final AttributeValidator attributeValidator,
+            @NonNull final ConsensusUpdateTopicTransactionBody op,
+            @NonNull final Topic topic) {
+        if (op.hasFeeScheduleKey()) {
+            validateTrue(topic.hasFeeScheduleKey(), FEE_SCHEDULE_KEY_CANNOT_BE_UPDATED);
+            if (!KeyOneOfType.UNSET.equals(op.feeScheduleKey().key().kind())) {
```

Comment:
If this is `UNSET` is it not error ?

Reply:
Aren't we supposed to remove the fee_schedule_key, also here we are checking the new key validity.

Reply:
The key will be removed only if the Key is `IMMUTABILITY_SENTINEL_KEY`, which has a default keyList. The same is checked in `isKeyRemoval`.

Reply:
I see, I guess we don't need this if statement at all then because the attributeValidator.validateKey that we are calling inside checks if the key is UNSET and if true throws BAD_ENCODING exception.